app/models/split.rb

```diff
@@ -48,12 +49,33 @@ class Split < ActiveRecord::Base
   end
 
   def self.arel_excluding_incomplete_features_for(app_build)
-    Arel::Nodes::Or.new(
-      arel_table[:feature_gate].eq(false),
-      AppFeatureCompletion.select(1).satisfied_by(app_build).arel.exists
+    Arel::Nodes::Grouping.new(
+      arel_table[:feature_gate].eq(false)
+      .or(
+        AppFeatureCompletion.select(1).satisfied_by(app_build).arel
+        .where(AppFeatureCompletion.arel_table[:split_id].eq(arel_table[:id]))
+        .exists
+      )
     )
   end
 
+  scope :with_remote_kill_knockouts_for, ->(app_build) do
+    previous_selects = all.arel.projections
+    except(:select)
+      .select(
+        previous_selects,
+        AppRemoteKill.select(:override_to).affecting(app_build).arel
```

Comment:
So the magic here is that you can rip out an `arel` node, adjust it a little, and then shove it straight into `.select()`? Fancy.